### PR TITLE
[rtsan][compiler-rt] Get rid of [[blocking]] stub in tests

### PR DESCRIPTION
**Pull request by cjappl**
Added the printfs to ensure we wouldn't have these empty functions compiled out.

### CODE REVIEW

File: compiler-rt/test/rtsan/blocking_call.cpp [hunk 1]
```diff
@@ -28,7 +21,7 @@ int main() {
   return 0;
   // CHECK-NOT: {{.*safe_call*}}
   // CHECK: ==ERROR: RealtimeSanitizer: blocking-call
-  // CHECK-NEXT: Call to blocking function `custom_blocking_function` in real-time context!
+  // CHECK-NEXT: Call to blocking function `custom_blocking_function()` in real-time context!
```
Comment (cjappl) [inline compiler-rt/test/rtsan/blocking_call.cpp:24]: This change was because __func__ behaves different from the LLVM IR demangling of the same func

**Comment by cjappl** [comment]
@davidtrevelyan I'm submitting this once tests are green, feel free to review before or post-commit

File: compiler-rt/test/rtsan/blocking_call.cpp [hunk 2]
```diff
@@ -7,18 +7,11 @@
...
-  __rtsan_notify_blocking_call(__func__);
+void custom_blocking_function() [[clang::blocking]] {
+  printf("In blocking function\n");
 }
 
-void safe_call() {
-  // TODO: When [[blocking]] is implemented, don't call this directly.
-  __rtsan_notify_blocking_call(__func__);
-}
+void safe_call() [[clang::blocking]] { printf("In safe call\n"); }
```
Comment (davidtrevelyan) [inline compiler-rt/test/rtsan/blocking_call.cpp:14]: I'm a bit confused as to why we have a function called `safe_call` that's attributed `[[clang::blocking]]`. The test assertions also check that `safe_call` is not printed. Was it originally a mistake to call `__rtsan_notify_blocking_call` in the test when it was first written, maybe?

File: compiler-rt/test/rtsan/blocking_call.cpp [hunk 3]
```diff
@@ -28,7 +21,7 @@ int main() {
   return 0;
   // CHECK-NOT: {{.*safe_call*}}
   // CHECK: ==ERROR: RealtimeSanitizer: blocking-call
-  // CHECK-NEXT: Call to blocking function `custom_blocking_function` in real-time context!
+  // CHECK-NEXT: Call to blocking function `custom_blocking_function()` in real-time context!
   // CHECK-NEXT: {{.*custom_blocking_function*}}
```
Comment (davidtrevelyan) [inline compiler-rt/test/rtsan/blocking_call.cpp:25]: Would it be wise to add `// CHECK-NOT: {{.*In blocking function.*}}` (or whatever the correct regex is to make sure the printf doesn't happen inside `custom_blocking_function`?

File: compiler-rt/test/rtsan/blocking_call.cpp (hunk 2 above)
Comment (cjappl) [reply compiler-rt/test/rtsan/blocking_call.cpp:14]: This call is to test that [[blocking]] called from an un-attributed setting (main) does not result in any error. Happy to take suggestions on how to make this more clear!

Comment (cjappl) [reply compiler-rt/test/rtsan/blocking_call.cpp:14]: I just took a second crack at it with different assertions etc, let me know what you think

Comment (davidtrevelyan) [reply compiler-rt/test/rtsan/blocking_call.cpp:14]: Oops - I missed the call to `safe_call` in `main` - my mistake. Anyway, if we wanted to make an improvement, I think we could organise `main` into:

```cpp
int main() {
    nonrealtime_function();
    realtime_function();
    return 0;
}
```

where the wrappers `nonrealtime_function` and `realtime_function` are simply:

```cpp
void realtime_function() [[clang::nonblocking]] { custom_blocking_function(); }
void nonrealtime_function() { custom_blocking_function(); }
```

and notice we have eliminated `safe_call`.

**Review by davidtrevelyan — APPROVED** [review]

**Comment by cjappl** [comment]
Build failure unrelated - merging

**Pull request merged**